### PR TITLE
Feature/expanded timeline

### PR DESCRIPTION
**Pull request by MarkRivera**
# Description

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/components/incidents/IncidentFilter.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { DateTime } from 'luxon';
+import Interval from 'luxon/src/interval.js';
+
+const newData = arr => {
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js]: While it's obvious that "arr" is an array, I think it's valuable to use descriptive and declarative names with your variables.

File: src/components/incidents/IncidentFilter.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const newCurrentData = arr.map(obj =>
+    Object.keys(obj)
+      .filter(x => obj[x] !== null)
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js]: Same comment as above. I know to you it was just a quick thing to use to get it functional, but that might not be clear to anyone coming behind you to maintain this code.

File: src/components/incidents/IncidentFilter.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      .reduce((o, e) => {
+        o[e] = obj[e];
+        return o;
+      }, {})
+  );
+  const noUndefined = newCurrentData.map(obj =>
+    Object.keys(obj)
+      .filter(x => obj[x] !== undefined)
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js]: Same as above.

File: src/components/incidents/IncidentFilter.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      .reduce((o, e) => {
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js]: Same as above.

File: src/components/incidents/IncidentFilter.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        o[e] = obj[e];
+        return o;
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js]: Same as above. I have zero clue what "o" is.

File: src/components/incidents/IncidentFilter.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  );
+  return noUndefined;
+};
+
+const filterDataByState = (data, state) => {
+  return data.filter(incident => incident.state === state);
+};
+
+const createRange = range => {
+  // End date is exclusive, so add one day so the user can include the day he selected
```
Comment (jlong5795) [inline src/components/incidents/IncidentFilter.js:32]: This is an example of a FANTASTIC comment!

File: src/components/incidents/IncidentsCard.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import { DateTime } from 'luxon';
+import { Card } from 'antd';
+import 'antd/dist/antd.css';
+
+const IncidentsCard = props => {
+  let cityState = `${props.incident.city}, ${props.incident.state}`;
+
+  return (
+    <div className="cardBox">
+      {/* for states, I was thinking we could import the abbreviations from the bargraphAssets file */}
```
Comment (jlong5795) [inline src/components/incidents/IncidentsCard.js:11]: This kind of comment is good to include in PR or PR draft for discussion. I'd limit these comments in the future.

**Review by jlong5795 — CHANGES_REQUESTED** [review]
Please look at comments and update variable declarations in a more semantic way.

File: src/components/incidents/IncidentFilter.js (hunk 1 above)
Comment (MarkRivera) [reply src/components/incidents/IncidentFilter.js]: All of this were left over from Anthony's code, I'm not sure what he's doing with this function. I'll get to altering variables to make this understandable

File: src/components/incidents/IncidentsCard.js (hunk 7 above)
Comment (jstewart8053) [reply src/components/incidents/IncidentsCard.js:11]: I will make sure to include those kind of comments in the PR next time, thank you.

**Review by jstewart8053 — APPROVED** [review]

**Comment by MarkRivera** [comment]
All comments have been addressed

**Review by jlong5795 — APPROVED** [review]
Thanks for working on those items.

**Pull request merged**